### PR TITLE
Make link to programmatic auth more obvious

### PR DESCRIPTION
**Pull request by mojito317**
## PR summary
The README had some links that pointed at external URLs about other authentication modes rather than env var configuration but it was not so obvious that includes the programmatic auth mode as well. The aim of this PR to highlight those links by creating a new section for them.

I also considered it is time to refactor the README file a little because the lines were getting longer and longer, and it was tough to read the lines at maintenance/review. The long links now also have a reference instead of inline linking. I was trying to keep the <80 char/line length, I broke this rule only if an inseparable link was in that line.

I put full stop where it was not before, and fixed some grammar issues.

I also found some markdown linting  and go fmt issues:
* missing whitespaces from the beginning of a line if the list item was broken down into 2 lines
* go fmt found unnecessary whitespace in the examples

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [ ] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [x] Documentation content changes
- [ ] Other (please describe)

## What is the current behavior?
The README had some links that pointed at external URLs about other authentication modes but it was not so obvious that includes the programmatic auth mode as well.

## What is the new behavior?
The README contains a new section that highlights the links of external configuration and the programmatic auth mode. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

-

### CODE REVIEW

**Review by ricellis — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -409,7 +467,7 @@ func main() {
 
 	// 2. Update the document ==============================================
 	exampleDbName := "orders"
-	exampleDocID  := "example"
+	exampleDocID := "example"
```
Comment (emlaver) [inline README.md:470]: Do we want to keep the extra space to line up the `:=` for both variables?

File: README.md [hunk 2]
```diff
@@ -500,7 +559,7 @@ func main() {
 
 	// 2. Delete the document ==============================================
 	exampleDbName := "orders"
-	exampleDocID  := "example"
+	exampleDocID := "example"
```
Comment (emlaver) [inline README.md:562]: Same as above

File: examples/src/create_db_and_doc/create_db_and_doc.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ func main() {
 
 	// 3. Create a document ================================================
 	// 3.1. Create a document object with "example" id
-	exampleDocID    := "example"
+	exampleDocID := "example"
```
Comment (emlaver) [inline examples/src/create_db_and_doc/create_db_and_doc.go:54]: Same as above

File: examples/src/delete_doc/delete_doc.go [hunk 4]
```diff
@@ -32,7 +32,7 @@ func main() {
 
 	// 2. Delete the document ==============================================
 	exampleDbName := "orders"
-	exampleDocID  := "example"
+	exampleDocID := "example"
```
Comment (emlaver) [inline examples/src/delete_doc/delete_doc.go:35]: Same as above

File: examples/src/update_doc/update_doc.go [hunk 5]
```diff
@@ -32,7 +32,7 @@ func main() {
 
 	// 2. Update the document ==============================================
 	exampleDbName := "orders"
-	exampleDocID  := "example"
+	exampleDocID := "example"
```
Comment (emlaver) [inline examples/src/update_doc/update_doc.go:35]: Same as above

**Review by emlaver — APPROVED** [review]
+1 - disregard my comments below if they fall under the linting issue "go fmt found unnecessary whitespace in the examples".

**Comment by mojito317** [comment]
>  disregard my comments below if they fall under the linting issue "go fmt found unnecessary whitespace in the examples".

Yeah, they fall under it.

Thanks for the reviews @emlaver and @ricellis! I'm now about to merge this

**Pull request merged**